src/sharepoint/rest/folders.ts

```diff
@@ -0,0 +1,135 @@
+"use strict";
+
+import { Queryable } from "./Queryable";
+import * as Util from "../../utils/util";
+import * as Mixins from "./mixins";
+import { Files } from "./files";
+import { Gettable, SelectableGettable } from "./actionables";
+
+/**
+ * Describes a collection of Folder objects
+ * 
+ */
+export class Folders extends Queryable implements Mixins.Gettable, Mixins.Selectable, Mixins.Filterable {
+
+    /**
+     * Creates a new instance of the Folders class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this fields collection
+     */
+    constructor(baseUrl: string | Queryable) {
+        super(baseUrl, "folders");
+    }
+
+    /**
+     * Gets a folder by folder name
+     * 
+     */
+    public getByName(name: string): Folder {
+        this.concat(`('${name}')`);
```

Comment:
Contenttype IDs are represented with quotes when returned as a string. Use the same for other strange string representations?